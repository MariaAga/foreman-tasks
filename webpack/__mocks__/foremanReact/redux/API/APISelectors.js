export const selectAPIResponse = (state, key) => ({
  data: {
    text: 'some-data',
    key,
  },
});

export const selectAPIStatus = (state, key) => 'PENDING';
export const selectAPIError = (state, key) => ({ error: `${key} ERRROR` });
