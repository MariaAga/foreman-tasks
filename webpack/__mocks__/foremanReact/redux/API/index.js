export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};

export const get = data => ({ type: 'some-type', ...data });
