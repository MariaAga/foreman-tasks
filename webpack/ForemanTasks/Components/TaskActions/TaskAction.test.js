import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/API';
import {
  cancelTaskRequest,
  resumeTaskRequest,
  forceCancelTaskRequest,
  unlockTaskRequest,
} from './';

jest.mock('foremanReact/components/common/table', () => ({
  getTableItemsAction: jest.fn(controller => controller),
}));

jest.mock('foremanReact/API');

const task = ['some-id', 'some-name'];

const fixtures = {
  'should cancelTaskRequest and succeed': () => cancelTaskRequest(...task),
  'should cancelTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return cancelTaskRequest(...task);
  },

  'should resumeTaskRequest and succeed': () => {
    API.post.mockImplementation(() => ({ data: 'some-data' }));
    return resumeTaskRequest(...task);
  },
  'should resumeTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return resumeTaskRequest(...task);
  },
  'should forceCancelTaskRequest and succeed': () => {
    API.post.mockImplementation(() => ({ data: 'some-data' }));
    return forceCancelTaskRequest(...task);
  },
  'should forceCancelTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return forceCancelTaskRequest(...task);
  },
  'should unlockTaskRequest and succeed': () => {
    API.post.mockImplementation(() => ({ data: 'some-data' }));
    return unlockTaskRequest(...task);
  },
  'should unlockTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return forceCancelTaskRequest(...task);
  },
};
describe('Tasks actions', () => {
  testActionSnapshotWithFixtures(fixtures);
});
