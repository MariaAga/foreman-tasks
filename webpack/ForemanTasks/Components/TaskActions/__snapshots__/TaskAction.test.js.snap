// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks actions should cancelTaskRequest and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Trying to cancel \\"some-name\\" task",
          "type": "info",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task cannot be cancelled at the moment.",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`Tasks actions should cancelTaskRequest and succeed 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Trying to cancel \\"some-name\\" task",
          "type": "info",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task cancelled",
          "type": "success",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`Tasks actions should resumeTaskRequest and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Task \\"some-name\\" has to be resumable.",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`Tasks actions should resumeTaskRequest and succeed 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task execution was resumed",
          "type": "success",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;
